src/main/java/org/mockbukkit/mockbukkit/block/data/decoder/PropertyDecoder.java

```diff
@@ -0,0 +1,14 @@
+package org.mockbukkit.mockbukkit.block.data.decoder;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import javax.annotation.Nullable;
+
+@ApiStatus.Internal
+public interface PropertyDecoder<T>
+{
+
+	@Nullable
+	T decode(@Nullable Object value);
```

Comment:
```suggestion
	T decode(@Nullable String value);
```
Easier to deal with if it's done like this